boa/src/object/mod.rs

```diff
@@ -976,90 +1001,33 @@ impl<'context> ConstructorBuilder<'context> {
 
         self.constructor_object.borrow_mut().insert_property(
             binding.binding,
-            function,
-            Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
+            PropertyDescriptor::builder()
+                .value(function)
+                .writable(true)
+                .enumerable(false)
+                .configurable(true),
         );
         self
     }
 
-    /// Add new data property to the constructor's prototype.
-    #[inline]
-    pub fn property<K, V>(&mut self, key: K, value: V, attribute: Attribute) -> &mut Self
-    where
-        K: Into<PropertyKey>,
-        V: Into<Value>,
-    {
-        let property = DataDescriptor::new(value, attribute);
-        self.prototype.borrow_mut().insert(key, property);
-        self
-    }
-
-    /// Add new static data property to the constructor object itself.
-    #[inline]
-    pub fn static_property<K, V>(&mut self, key: K, value: V, attribute: Attribute) -> &mut Self
-    where
-        K: Into<PropertyKey>,
-        V: Into<Value>,
-    {
-        let property = DataDescriptor::new(value, attribute);
-        self.constructor_object.borrow_mut().insert(key, property);
-        self
-    }
```

Comment:
I think methods like `.property` always expect a data descriptor so we should still take `Attribute` and in the function construct a data descriptor.

Reply:
I tried to preserve Attribute but I realized there's not a good way to say something like "`[[writable]]` is not `false` but `None`"  unless I pass several `Option<bool>` as arguments, so I decided to pass the whole `PropertyDescriptorBuilder`. In addition to that, you can pass an accessor descriptor without needing another separate function, making the creation of new properties a little bit less convoluted.

Reply:
> I tried to preserve `Attribute` but I realized there's not a good way to say something like "`[[writable]]` is not `false` but `None`" unless I pass several `Option<bool>` as arguments, so I decided to pass the whole `PropertyDescriptorBuilder` instead. In addition to that, you can pass an accessor descriptor without needing another separate function, making the creation of new properties a little bit more simple.

`.property` should always create a complete data descriptor, same goes for accessor but for accessor descriptor. If we want to add a property descriptor we can call `.property_descriptor`, this way the API is more intuitive, since this is exposed to the user.

Reply:
Okay, let me fix this then

Reply:
Reverted my api changes. The number of passed tests is the same as before so it should behave correctly.